app/workers/subject_priority_worker.rb

```diff
@@ -0,0 +1,13 @@
+class SubjectPriorityWorker
+  include Sidekiq::Worker
+
+  def perform(sms_ids)
+    sms_ids.each do |sms_id|
```

Comment:
A lot of small updates can be bad for the db, especially if it has to update indexes though this col does not have one.

Ideally this would do batch updates in 1 sql transaction instead of issuing lots of small AR save sql statements. Sadly in our rails / pg version there isn't a great way to do this.

PG 9.5 added upserts - https://wiki.postgresql.org/wiki/UPSERT

Active record import (we use this) supports this but only for PG >= 9.5 - https://github.com/zdennis/activerecord-import/#duplicate-key-update

Rails 6 natively supports this (might be worth adding a comment explaining we should replace code in future) - https://edgeapi.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-upsert

https://www.oipapio.com/question-751809
I'm not sure but you might be able to use `update_all` with a select statement on the linked subjects table. If you can get that working that will be the best way, e.g.
```
SetMemberSubject.update_all('priority = (SELECT subjects.metadata->>'#priority' FROM subjects WHERE set_member_subjects.subject_id = subjects.id AND set_member_subejct.id = ?)', sms_ids)
```
I'd be happy to help you work on this query too

Otherwise the sub-optimal solution is to batch up your saves and wrap them in a transaction - https://stackoverflow.com/a/19037842/2761423

I'd be super interested to see any perf improvements between the current example and the final implementation https://blog.appsignal.com/2018/02/27/benchmarking-ruby-code.html

 FWIW - we currently spam the db when creating sws records on subject set changes, e.g. https://github.com/zooniverse/Panoptes/blob/414baf509f85a569b388bd47e16122ee253217bc/app/workers/subject_set_statuses_create_worker.rb#L21 

That needs to change as well, ideally using a batchable solution to cut down the db pressure and not just relying on time spacing the operations out.

Reply:
This is great, thanks! Exactly what I was hoping to talk about. No rush on this, I mainly wanted to get a dumb version out to start this conversation. Lots of potential pain points, this seems like something to pair on sometime? Thanks for the research, I'll go read some blog posts.